core/src/utils/transition/ios.transition.ts

```diff
@@ -123,56 +145,131 @@ const animateBackButton = (
   rootAnimation: Animation,
   rtl: boolean,
   backDirection: boolean,
-  backButtonEl: any,
-  largeTitleBox: DOMRect,
-  backButtonBox: DOMRect
+  backButtonEl: HTMLIonBackButtonElement,
+  backButtonBox: DOMRect,
+  backButtonTextEl: HTMLElement,
+  backButtonTextBox: DOMRect,
+  largeTitleEl: HTMLIonTitleElement,
+  largeTitleTextBox: DOMRect
 ) => {
   const BACK_BUTTON_START_OFFSET = rtl ? `calc(100% - ${backButtonBox.right + 4}px)` : `${backButtonBox.left - 4}px`;
-  const START_TEXT_TRANSLATE = rtl ? '7px' : '-7px';
-  const END_TEXT_TRANSLATE = rtl ? '-4px' : '4px';
-
-  const ICON_TRANSLATE = rtl ? '-4px' : '4px';
 
   const TEXT_ORIGIN_X = rtl ? 'right' : 'left';
   const ICON_ORIGIN_X = rtl ? 'left' : 'right';
 
+  const CONTAINER_ORIGIN_X = rtl ? 'right' : 'left';
+
+  /**
+   * When the title and back button texts match
+   * then they should overlap during the page transition.
+   * If the texts do not match up then the back button text scale adjusts
+   * to not perfectly match the large title text otherwise the
+   * proportions will be incorrect.
+   */
+  const doTitleAndButtonTextsMatch = backButtonTextEl.textContent === largeTitleEl.textContent;
+
+  const WIDTH_SCALE = largeTitleTextBox.width / backButtonTextBox.width;
+
+  /**
+   * We subtract 10px to account for slight sizing/padding
+   * differences between the title and the back button.
+   */
+  const HEIGHT_SCALE = (largeTitleTextBox.height - LARGE_TITLE_SIZE_OFFSET) / backButtonTextBox.height;
+
+  const TEXT_START_SCALE = doTitleAndButtonTextsMatch
+    ? `scale(${WIDTH_SCALE}, ${HEIGHT_SCALE})`
```

Comment:
Could we have a comment explaining that we also need to scale by width to account for the different line weights? That had been the biggest question in my head before you showed precisely how the animation breaks down.

Reply:
Added in https://github.com/ionic-team/ionic-framework/pull/28290/commits/f418f6b04e7b097327ef24b0b79e434eaa6e5ff4